src/pytest_inmanta_lsm/load_generator.py

```diff
@@ -0,0 +1,199 @@
+"""
+    Pytest Inmanta LSM
+
+    :copyright: 2024 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+
+import asyncio
+import datetime
+import logging
+import threading
+import time
+
+from pytest_inmanta_lsm import remote_orchestrator
+
+LOGGER = logging.getLogger(__name__)
+
+
+# Custom Thread Class
+class LoadThread(threading.Thread):
```

Comment:
1. I would make the exception behavior to log it and keep going.  Perhaps keep a boolean if the run was correct for the testcase to assert on
2. I would just put a try except block around the actual code, instead of this fairly complicated custom thread